impeller/tools/malioc_diff.py

```diff
@@ -277,6 +299,16 @@ def compare_shaders(malioc_tree, before_shader, after_shader):
 
 def main(argv):
   args = parse_args(argv[1:])
+
+  # Generate full paths if relative ones are provided with before and
+  # after taking precedence.
+  args.before = (
```

Comment:
How does this fail when neither `--before` nor `--before-relative-to-src` is provided. (Same question for the `after` versions.) Maybe putting the fallback logic in the argument validation in `validate_args` would be more clear?

Reply:
Moved it to the validation method and added some extra validations.